dms-service/src/main/java/de/muenchen/oss/swim/dms/application/usecase/ProcessFileUseCase.java

```diff
@@ -0,0 +1,142 @@
+        log.info("Processing file {} for use case {}", file, useCaseName);
+        final UseCase useCase = swimDmsProperties.findUseCase(useCaseName);
+        log.debug("Resolved use case: {}", useCase);
+        // load file
+        try (InputStream fileStream = fileSystemOutPort.getPresignedUrlFile(presignedUrl)) {
+            // get target coo
+            final DmsTarget dmsTarget = resolveTargetCoo(metadataPresignedUrl, useCase, file);
+            log.debug("Resolved dms target: {}", dmsTarget);
+            // get filename
+            final String filename = this.applyOverwritePattern(useCase.getFilenameOverwritePattern(), file.getFileName(), PATTERN_JOINER);
+            // transfer to dms
+            switch (useCase.getType()) {
+            // to dms inbox
+            case INBOX -> dmsOutPort.putFileInInbox(dmsTarget, filename, fileStream);
+            // create dms incoming
+            case INCOMING_OBJECT -> {
+                // resolve name for IncomingObject
+                final String incomingObjectName;
+                if (Strings.isBlank(useCase.getContentObjectNamePattern())) {
+                    // use overwritten filename if no pattern for IncomingObject name is defined
+                    incomingObjectName = filename;
+                } else {
+                    // else apply pattern to original filename
+                    incomingObjectName = this.applyOverwritePattern(useCase.getContentObjectNamePattern(), file.getFileName(), PATTERN_JOINER);
+                }
+                // create Incoming
+                dmsOutPort.createIncoming(dmsTarget, filename, incomingObjectName, fileStream);
+            }
+            }
+        } catch (final IOException e) {
+            throw new PresignedUrlException("Error while handling file InputStream", e);
+        }
+        // mark file as finished
+        fileEventOutPort.fileFinished(useCaseName, presignedUrl, metadataPresignedUrl);
+    }
+
+    /**
+     * Resolve target coo for useCase.
+     * {@link UseCase.Type}
+     *
+     * @param metadataPresignedUrl Presigned url for metadata file.
+     * @param useCase The use case.
+     * @return The resolved coo.
+     */
+    protected DmsTarget resolveTargetCoo(final String metadataPresignedUrl, final UseCase useCase, final File file)
+            throws MetadataException, PresignedUrlException {
+        return switch (useCase.getCooSource()) {
+        case METADATA_FILE -> this.resolveMetadataTargetCoo(metadataPresignedUrl, useCase);
+        // TODO resolve coo from filename
+        case FILENAME -> throw new UnsupportedOperationException("Coo source type filename not implemented yet");
+        case STATIC -> new DmsTarget(useCase.getTargetCoo(), useCase.getUsername(), useCase.getJoboe(), useCase.getJobposition());
+        case OU_DEFAULT -> new DmsTarget(null, useCase.getUsername(), useCase.getJoboe(), useCase.getJobposition());
```

Comment:
Warum hast du das OU_DEFAULT genannt? Was für ein Default ist das? Es landet doch in dem Fall im Arbeitsvorrat des Users in der angegebenen Stelle. Was ist denn der Fabasoft-englische Begriff für Arbeitsvorrat?

Reply:
> Was ist denn der Fabasoft-englische Begriff für Arbeitsvorrat

Hab ich leider nicht gefunden, das Wort "Arbeitsvorrat" kommt auch nur ein einziges mal in der ganzen Doku vor.
Können wir gerne noch umbenennen.
`ou_default` hab ich das genannt, weil der Arbeitsvorrat für mich einfach nur ein default Vorgang der OU ist.

Reply:
OK, lassen wir es erstmal so. Aber ich glaube nicht, dass der Arbeitsvorrat ein Vorgang ist, demzufolge finde ich das JavaDoc in Usecase zu INCOMING_OBJECT auch nicht richtig. Wir sollten für den Arbeitsvorrat einen anderen Begriff nutzen, notfalls denken wir uns einen aus (wie wäre "work queue"?)

Reply:
Mach ich im Nachgang